### PR TITLE
feat(compute): samples/compute_instances_bulk_insert

### PR DESCRIPTION
**Pull request by irudykss**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/instances/create-start-instance/create_instance_test.go [hunk 1]
```diff
@@ -199,3 +199,100 @@ func TestComputeCreateInstanceSnippets(t *testing.T) {
...
+		Project:          tc.ProjectID,
+		InstanceTemplate: instanceTemplateName,
+	}
+
+	op, err = instanceTemplatesClient.Delete(ctx, deleteTemplateReq)
+	if err != nil {
+		t.Errorf("unable to delete instance template: %v", err)
+	}
+
+	if err = op.Wait(ctx); err != nil {
```
Comment (muncus) [inline compute/instances/create-start-instance/create_instance_test.go:295]: no change necessary for this, but in the future, i dont think its necessary for us to wait for the cleanup operation at the end of the tests.

File: compute/instances/create-start-instance/create_instance_test.go [hunk 2]
```diff
@@ -199,3 +199,100 @@ func TestComputeCreateInstanceSnippets(t *testing.T) {
...
+
+	namePattern := "i-#-" + fmt.Sprint(seededRand.Int())[0:5]
+	buf := &bytes.Buffer{}
+
+	_, err = createFiveInstances(buf, tc.ProjectID, zone, instanceTemplateName, namePattern)
+	if err != nil {
+		t.Errorf("createFiveInstances got err: %v", err)
+	}
+
+	for i := 1; i <= 5; i++ {
```
Comment (muncus) [inline compute/instances/create-start-instance/create_instance_test.go:277]: the `createFiveInstances` function returns the Instance objects. we could just iterate over that list and delete those instances directly. that avoids the need for the string replacement in the pattern.

File: compute/instances/create-start-instance/create_instance_test.go [hunk 3]
```diff
@@ -199,3 +199,100 @@ func TestComputeCreateInstanceSnippets(t *testing.T) {
...
+	}
+	defer instanceTemplatesClient.Close()
+
+	var seededRand *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "europe-central2-b"
+	instanceTemplateName := "test-instance-template" + fmt.Sprint(seededRand.Int())
+	machineType := "n1-standard-1"
+	sourceImage := "projects/debian-cloud/global/images/family/debian-10"
```
Comment (muncus) [inline compute/instances/create-start-instance/create_instance_test.go:231]:
```suggestion
	sourceImage := "projects/debian-cloud/global/images/family/debian-12"
```
debian-10 is near its end of life. using a newer release will buy us more time before we need to update these tests.

File: compute/instances/create-start-instance/create_instance_test.go [hunk 4]
```diff
@@ -199,3 +199,100 @@ func TestComputeCreateInstanceSnippets(t *testing.T) {
 		t.Errorf("deleteDisk got err: %v", err)
 	}
 }
+
+func TestComputeBulkCreateInstanceSnippets(t *testing.T) {
+	//ctx := context.Background()
+	//var r *rand.Rand = rand.New(
+	//	rand.NewSource(time.Now().UnixNano()))
+	//tc := testutil.SystemTest(t)
```
Comment (muncus) [inline compute/instances/create-start-instance/create_instance_test.go:207]: please be sure to call `testutil.SystemTest()`, as there are some environment variable checks in there that we want consistently applied to all tests.

it looks like the rest of this was for reference, and can probably be deleted.

File: compute/instances/create-start-instance/create_instance_test.go (hunk 1 above)
Comment (irudykss) [reply compute/instances/create-start-instance/create_instance_test.go:295]: I will touch PR anyway, so i've removed it 👍

File: compute/instances/create-start-instance/bulk_insert.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+	//     with each character corresponding to one digit of the generated instance
+	//     name. Example: a namePattern of inst-#### generates instance names such
+	//     as inst-0001 and inst-0002. If existing instances in the same project and
+	//     zone have names that match the name pattern then the generated instance
+	//     numbers start after the biggest existing number. For example, if there
+	//     exists an instance with name inst-0050, then instance names generated
+	//     using the pattern inst-#### begin with inst-0051. The name pattern
+	//     placeholder #...# can contain up to 18 characters.
+
+	template, err := getInstanceTemplate(projectID, templateName)
```
[the remark this answers is not in the document]
Comment (irudykss) [reply compute/instances/create-start-instance/bulk_insert.go:126]: Sure, done ✌️

File: compute/instances/create-start-instance/bulk_insert.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+	return instances, nil
+}
+
+func createFiveInstances(w io.Writer, projectID, zone, templateName, namePattern string) ([]*computepb.Instance, error) {
+	// projectID := "your_project_id"
+	// zone := "us-central1-a"
+	// templateName := "your_instance_template_name"
+	// namePattern := "instance-####"
+
+	// 	   namePattern NOTE: The string pattern used for the names of the VMs. The pattern
```
[the remark this answers is not in the document]
Comment (irudykss) [reply compute/instances/create-start-instance/bulk_insert.go:115]: ok, removed ✌️

File: compute/instances/create-start-instance/bulk_insert.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/google/uuid"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+func bulkInsertInstance(w io.Writer, projectID, zone string, template *computepb.InstanceTemplate, count int64, namePattern string, minCount *wrapperspb.Int64Value, labels map[string]string) ([]*computepb.Instance, error) {
```
[the remark this answers is not in the document]
Comment (irudykss) [reply compute/instances/create-start-instance/bulk_insert.go:31]: got you, removed 👍

File: compute/instances/create-start-instance/create_instance_test.go (hunk 3 above)
Comment (irudykss) [reply compute/instances/create-start-instance/create_instance_test.go:231]: 👍

File: compute/instances/create-start-instance/create_instance_test.go (hunk 4 above)
Comment (irudykss) [reply compute/instances/create-start-instance/create_instance_test.go:207]: it is leftovers, removed

File: compute/instances/create-start-instance/bulk_insert.go [hunk 8]
```diff
@@ -28,37 +28,47 @@ import (
...
+		Count:                  proto.Int64(5),
 		MinCount:               proto.Int64(minCount.GetValue()),
 		NamePattern:            &namePattern,
 	}
 
-	if labels == nil {
-		labels = make(map[string]string)
-	}
-
+	labels := make(map[string]string)
```
Comment (muncus) [inline compute/instances/create-start-instance/bulk_insert.go:71]: you can inline this map as follows:
```suggestion
	labels := map[string]string{
	  "bulk_batch": uuid.New().String()
	}
```

**Review by muncus — APPROVED** [review]

**Pull request merged**